debug3.py

```diff
@@ -0,0 +1,462 @@
+            op = op_nlo + op_nnlo
+
+    print(cc1 / (op @ f))
+
+
+check_c_res()
+# check B EKO is K
+print("check B EKO result")
+if order_mode == Order.nnlo:
+    ekob = output.EKO.load("data/ekos/2405/test-nnlo.tar")
+else:
+    ekob = legacy.load_tar("data/ekos/2205/test.tar")
+fb = apply_pdf(ekob, pdf)
+if mode_g:
+    gb = Kgg @ f
+    cb = Kqg @ f
+else:
+    gb = Kgq @ f
+    cb = Kqq @ f
+print(fb[300.0]["pdfs"][21] / gb)
+print(fb[300.0]["pdfs"][4] / cb)
+
+# check C EKO is identity
+print("check C EKO result")
+if order_mode == Order.nnlo:
+    ekoc = output.EKO.load("data/ekos/3405/test-nnlo.tar")
+else:
+    ekoc = legacy.load_tar("data/ekos/3205/test.tar")
+fc = apply_pdf(ekoc, pdf)
+if mode_g:
+    print(fc[2.0**2 * 300.0]["pdfs"][21] / f)
+    print(fc[2.0**2 * 300.0]["pdfs"][4] - np.zeros_like(f))
+else:
+    print(fc[2.0**2 * 300.0]["pdfs"][4] / f)
+    print(fc[2.0**2 * 300.0]["pdfs"][21] - np.zeros_like(f))
+
+# check B result
+print("check B result")
+if mode_g:
+    print(bb1 / (((loq + ab * nloq) @ Kqg + (ab * nlog) @ Kgg) @ f))
+    # plt.plot(bb1, label="b-fk")
+    # plt.plot(((loq + ab * nloq) @ (Kqg) + (ab * nlog) @ (Kgg)) @ f, label="ana_pred")
+else:
+    print(bb1 / (((loc + ab * nloc) @ Kqq + (ab * nlog) @ Kgq) @ f))
+    # plt.plot(bb1, label="b-fk")
+    # plt.plot(((loc + ab * nloc) @ Kqq + (ab * nlog) @ Kgq) @ f, label="ana_pred")
+# plt.legend()
+# plt.show()
+# check difference between B and C
+print("check B - C")
+if order_mode == Order.nlo or order_mode == Order.nnlo:
+    diff_grid = bb1 - cc1
+    plt.plot(bb1, label="b-grid")
+    plt.plot(cc1, label="c-grid")
+    plt.legend()
+    plt.show()
+    if mode_g:
+        # plt.plot(ab**2 * L * ((nloc @ pqg0 @ f)), label="pqg0")
+        # plt.plot(ab**2 * L * ((nlog @ pgg0 @ f)), label="pgg0")
+        if order_mode == Order.nlo:
+            diff_ana = (ab * ab * L * ((nloq @ pqg0) + (nlog @ pgg0))) @ f
+            if signed:
+                diff_ana = (
+                    -2 * ab * L * (loq @ pqg0)
+                    - ab * ab * L * ((nloq @ pqg0) + (nlog @ pgg0))
+                ) @ f
+        else:
+            nnlomix = (
+                nlog @ pgg0 @ (pgg0 - beta0I)
+                + nlog @ pgq0 @ (pqg0 - beta0I)
+                + nloq @ pqq0 @ (pqg0 - beta0I)
+                + nloq @ pqg0 @ (pgg0 - beta0I)
+            )
+            nnlomix2 = (
+                nnlog @ pgg0 @ (pgg0 - beta0I)
+                + nnlog @ pgq0 @ (pqg0 - beta0I)
+                + nnloq @ pqq0 @ (pqg0 - beta0I)
+                + nnloq @ pqg0 @ (pgg0 - beta0I)
+            )
+            diff_ana_nnlo = (
+                ab
+                * ab
+                * ab
+                * (
+                    L
+                    * ((nloq @ pqg1) + (nlog @ pgg1) + (nnlog @ pgg0) + (nnloq @ pqg0))
+                    + L * L * ((-1.0 / 2.0) * nnlomix)
+                )
+                + ab
```

Comment:
you can trick `black` by using parenthesises: `(ab * ab * ab * ab)` which he will not break (and looks much better)